py/desisim/survey_release.py

```diff
@@ -247,71 +276,180 @@ def assign_exposures(self,exptime=None):
             if self.data is None:
                 raise ValueError("No data catalog nor exptime was provided. Please provide one of them")
             log.info("Assigning exposures")
-            filename='/global/cfs/cdirs/desi/users/hiramk/desi/quickquasars/sampling_tests/npass_pixmap.fits'
-            pixmap=Table.read(filename)
             mock=self.mockcatalog
             if 'TARGET_DEC' in self.data.colnames and 'TARGET_RA' in self.data.colnames:
-                data_pixels = hp.ang2pix(1024,np.radians(90-self.data['TARGET_DEC']),np.radians(self.data['TARGET_RA']),nest=True)
+                data_pixels = radec2pix(ra=self.data['TARGET_RA'], dec=self.data['TARGET_DEC'], nside=1024)
             elif 'DEC' in self.data.colnames and 'RA' in self.data.colnames: 
-                data_pixels = hp.ang2pix(1024,np.radians(90-self.data['DEC']),np.radians(self.data['RA']),nest=True)
+                data_pixels = radec2pix(ra=self.data['RA'], dec=self.data['DEC'], nside=1024)
             else:
                 raise ValueError("No RA,DEC or TARGET_RA,TARGET_DEC columns in data catalog")
                 
-            is_lya_data=self.data['Z']>2.1    
-            effective_exptime = lambda tsnr2_lrg: 12.25*tsnr2_lrg
-            exptime_data = effective_exptime(self.data['TSNR2_LRG'])
-            
+            is_lya_data=self.data['Z']>2.1
+            is_data_qsotgt=get_qsotarget_mask(self.data)
+            data_mask = is_lya_data&is_data_qsotgt
+            if 'TSNR2_LRG' in self.data.colnames:
+                log.info('Getting effective exposure time in data catalog by 12.15*TSNR2_LRG.')
+                exptime_data = 12.15*self.data['TSNR2_LRG']
+            elif 'TSNR2_LYA' in self.data.colnames:
+                log.info('Getting effective exposure time in data catalog by 11.8*TSNR2_LYA.')
+                exptime_data = 11.8*self.data['TSNR2_LYA']
+            elif 'TSNR2_QSO' in self.data.colnames:
+                log.info('Getting effective exposure time in data catalog by 33.61*TSNR2_QSO.')
+                exptime_data = 33.61*self.data['TSNR2_QSO']
+            else: 
+                raise ValueError("Can't compute effective exposure time. Data catalog columns should include TSNR2_LRG, TSNR2_LYA or TSNR2_QSO.")
```

Comment:
What is the assumption for `self.data` here? `TSNR2_LRG` has the highest priority, so should you raise a big warning if that is not found? Or are you imagining that the absence of `TSNR2_LRG` column is deliberate in order to run some tests using other columns, so a warning is not required?

Reply:
I added a warning if `TSNR2_LRG` is not in the data catalog. 

We use TSNR2_LRG as this is what the spectroscopic pipeline uses to define the effective exposure time. However  I've found some catalogs without this column so I added the option to use other target templates SNR.

Maybe @julienguy can provide more insight here.